magefile.go

```diff
@@ -1576,17 +1578,37 @@ func (Integration) Single(ctx context.Context, testName string) error {
 // UpdateVersions runs an update on the `.agent-versions.json` fetching
 // the latest version list from the artifact API.
 func (Integration) UpdateVersions(ctx context.Context) error {
-	// test 2 current 8.x version, 1 previous 7.x version and 1 recent snapshot
+	maxSnapshots := 3
+
+	branches, err := git.GetReleaseBranches(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to list release branches: %w", err)
+	}
+
+	// -1 because we manually add 7.17 below
+	if len(branches) > maxSnapshots-1 {
+		branches = branches[:maxSnapshots-1]
+	}
+
+	// it's not a part of this repository, cannot be retrieved with `GetReleaseBranches`
+	branches = append(branches, "7.17")
+
+	// uncomment if want to have the current version snapshot on the list as well
+	// branches = append([]string{"master"}, branches...)
```

Comment:
It's not necessary to have the current version snapshot on this list because we have plenty of tests that use it for upgrade/downgrade already. So, I skipped it to make the tests run faster.